Manual merge of first 7 tasks into develop

JPTFL opened the pull request:
Manually merges first 7 tasks into space station scene:
- mixing ingredients
- chemical disaster
- marbel gravity
- book riddle
- energy core
- storage task
- asteroid shooter

Manually copies related unity game objects from duplicated space station scene to main space station scene.
Refactors task code to match latest game code and space station structure.

You may checkout the develop_merge branch and see, if the `SpaceStation` scene and the integrated tasks work correctly in another workspace setup.

MarianK99 committed:
added puzzle asteroid-shooter

Aremju committed:
Merge branch 'develop' into puzzles_asteroidshooter

# Conflicts:
#	Assets/Scenes/SpaceStation.unity

Aremju committed:
merged develop into this branch

JudiTeller committed:
first iteration of integrity indicator

JudiTeller committed:
commented out method call

MarianK99 committed:
resized canvas to fit larger screen and added seperated inputs for each profile

JudiTeller committed:
Made HUD elements reachable through main HUD script

JudiTeller committed:
initial navigator setup

JudiTeller committed:
merge

JPTFL committed:
implements asteroid shooting game into game task system

JPTFL committed:
fitting canvas to window size

JPTFL committed:
changed using camera to using playerprofileservice

JudiTeller committed:
added marker calculations

JudiTeller committed:
hooked UI navigator into Task creation

JudiTeller committed:
updated spawn testscene

JPTFL committed:
seperated keyboard and vr crosshair movement

JPTFL committed:
first approach on structure for implementing the mixing ingredients task
- adds factory and task class
- adds first prefabs
- adds test materials

MarianK99 committed:
Added new method to start task with vr profile

MarianK99 committed:
possible solution to manage crosshair movement in vr

MarianK99 committed:
added canvas layer mask to layers and added it to crosshair canvas

MarianK99 committed:
changed layer of crosshair-canvas to "canvas"

MarianK99 committed:
added task explanation to ...\docs\puzzle\templates\Space

MarianK99 committed:
added basics (boxes and delivery plaform for riddle) with scripts

MarianK99 committed:
added basic task logic

MarianK99 committed:
added basic task lgoic

MarianK99 committed:
changed crosshair movement in vr from not working raycasting to using joystick

MarianK99 committed:
deactivated player movement during task (crosshair and player movement on the same joystick)

MarianK99 committed:
code refactoring

MarianK99 committed:
added missing reference to prefab

MarianK99 committed:
fixed missing if-statement

MarianK99 committed:
fixed some bugs

MarianK99 committed:
added storage-riddle prefab

MarianK99 committed:
Merge remote-tracking branch 'main-repo/puzzles_storage_room' into puzzles_storage_room

# Conflicts:
#	Assets/Scripts/Game/Tasks/StorageRiddle/HandleBoxDelivery.cs
#	Assets/Scripts/Game/Tasks/StorageRiddle/SpawnBoxes.cs
#	Assets/Scripts/Game/Tasks/StorageRiddle/StartStorageRiddle.cs

MarianK99 committed:
added game task structure to scene

MarianK99 committed:
added storage riddle to game tasks

MarianK99 committed:
Adding comments and did refactoring on code

JPTFL committed:
adds liquid and glass materials

JPTFL committed:
fixes issue, that initial difficulty is not set correctly, because constructor is called before updating SerializeField in unity editor
- adds to percentage conversion method

JPTFL committed:
adds first working approach for mixing ingredient task
- implements task factory/spawn points into duplicated space station scene MixingIngredientsTest for testing purpose
- needs to be tested in vr
- removes old unused test materials

MarianK99 committed:
added basic elements of this task

MarianK99 committed:
added energy core test scene

JPTFL committed:
adds valve and bar materials

JPTFL committed:
adds valve script, valve and general game task prefab

JPTFL committed:
adds space station duplicate test scene and adds test valve prefab instance

JPTFL committed:
adds approach to use explicit rotator object, which is used to rotate the valve
- needs to be tested in vr

JPTFL committed:
adds missing change for last commit

JPTFL committed:
adds a shuffle extension method for lists

JPTFL committed:
adds data class for task parameters

JPTFL committed:
adds medical disaster task and factory class
- adds valve attachment prefab, that is used to connect the valve to the pipes
- changes valve script to call event function when finished rotation goal

JPTFL committed:
adds medical task to duplicated space station test scene

JPTFL committed:
adds missing class documentation

juliusemilarendt-kb committed:
add test scene for marble gravity

juliusemilarendt-kb committed:
add labyrinth_textures

juliusemilarendt-kb committed:
add MarbleGravityFactory.cs

juliusemilarendt-kb committed:
implement marble riddle scripts

juliusemilarendt-kb committed:
update WallSwitcher.cs:
 - Will now destroy the switch and the wall

juliusemilarendt-kb committed:
Instantiate riddle

juliusemilarendt-kb committed:
make object return real game Task

juliusemilarendt-kb committed:
optimize imports

juliusemilarendt-kb committed:
add medium marble and test spacestation

juliusemilarendt-kb committed:
marble gravity prototype that works

juliusemilarendt-kb committed:
add easy labyrinth-texture

juliusemilarendt-kb committed:
add easy prefab

Aremju committed:
fixed crosshair movement and changed endpoint

juliusemilarendt-kb committed:
add marble gravity with vr

juliusemilarendt-kb committed:
add distance condition to riddle

Aremju committed:
tested marble gravity successfully

juliusemilarendt-kb committed:
Merge remote-tracking branch 'origin/marble_gravity' into marble_gravity

# Conflicts:
#	Assets/Scripts/Game/Tasks/marble_gravity/MarbleGravity.cs

juliusemilarendt-kb committed:
update marble gravity: fix compile errors

JPTFL committed:
adds grab anchor to bottle and allows to spawn liquid by controller activate event
- changes watering bottle nearby check to distance calculation, because OnTriggerEnter in WateringBottleDetection is not working when object is grabbed in vr, distance calculation needs to be tested

JPTFL committed:
add xr knob to valve
- needs to be tested, if the screw mechanic can be implemented using the xr know script

juliusemilarendt-kb committed:
change Texture

MarianK99 committed:
added puzzle energy core

JudiTeller committed:
changes to show marker on VR HUD

Aremju committed:
adjusted positions in HUD and marker clamp position

juliusemilarendt-kb committed:
add book_riddle_easy with 8 spawnpoints

JudiTeller committed:
changed UI animations

juliusemilarendt-kb committed:
add prefabs and necessary scripts

juliusemilarendt-kb committed:
finish marble gravity

JPTFL committed:
adds rotating valve by right controller rotation value
- needs to be tested in vr

MarianK99 committed:
added missing game logic

MarianK99 committed:
added other material

JPTFL committed:
adds working controller rotation to valve

JPTFL committed:
refactors valve script and adds dorcumentation

JPTFL committed:
corrects rotation direction to match controller rotation direction
- adds progress gradient

juliusemilarendt-kb committed:
add book prefab text

JPTFL committed:
Merge branch 'develop' into task_medical_disaster

JPTFL committed:
implements player profile reference from game task after merging develop
- adjusts valve points in prefab
- removes unused rotator script

JPTFL committed:
adds missing player profile service reference
- adds method to determine rotation transform

JPTFL committed:
changes rotation tracking to global rotation instead of local, to prevent incorrect rotation, when player turned or rotated

JPTFL committed:
changes movement type to have correct collider for flasks

JPTFL committed:
changes liquid to use simple standard material

JPTFL committed:
changed some task logic

MarianK99 committed:
collision handler outside of main script

juliusemilarendt-kb committed:
change scripts & prefab

juliusemilarendt-kb committed:
add debug message

juliusemilarendt-kb committed:
add new spawnpoint

juliusemilarendt-kb committed:
Merge branch 'develop' of github-personal.com:JudiTeller/S5-VR-game into book_riddle

juliusemilarendt-kb committed:
book spawn successful

Aremju committed:
finished energyCore-task

Aremju committed:
changed color of projectile

juliusemilarendt-kb committed:
add materials for books

juliusemilarendt-kb committed:
add materials to scene

juliusemilarendt-kb committed:
make random-effect more significant

juliusemilarendt-kb committed:
remove redundant comments

juliusemilarendt-kb committed:
finish book_riddle

juliusemilarendt-kb committed:
add missing doc

juliusemilarendt-kb committed:
fix desk

juliusemilarendt-kb committed:
add spawnpoints for marble_gravity

juliusemilarendt-kb committed:
refactor method

MarianK99 committed:
changed initialTimerTime to 600 seconds

JPTFL committed:
improves spawn points, liquid detection and ingredient size

JPTFL committed:
adds color progress indicator for mixing container
- removes unused code

juliusemilarendt-kb committed:
update book riddle prefab

JPTFL committed:
finish book riddle

JPTFL committed:
Merge remote-tracking branch 'origin/task_mixing_ingredients' into develop_merge

JPTFL committed:
Merge remote-tracking branch 'origin/task_medical_disaster' into develop_merge

JPTFL committed:
Merge remote-tracking branch 'origin/marble_gravity' into develop_merge

# Conflicts:
#	Assets/Prefabs/Player/PlayerProfile.prefab

JPTFL committed:
Merge remote-tracking branch 'origin/book_riddle' into develop_merge

JPTFL committed:
merged 4 riddles into space station (mixing ingredients, medical disaster, book_riddle, marble_gravity)

JPTFL committed:
remove unneccessary logging-messages

JPTFL committed:
removes unused empty game object "Spawnpoints" as the new structure after the merge intends to store the spawn points for a factory as children of the factory

JPTFL committed:
refactors unused debug log and print statements

JPTFL committed:
moves asteroid shooter scene to new file and reverts space station scene to current develop state

JPTFL committed:
moves energy core scene to new file and reverts space station scene to current develop state

JPTFL committed:
moves storage task scene to new file and reverts space station scene to current develop state

JPTFL committed:
Merge branch 'puzzles_energycore' into develop_merge

# Conflicts:
#	Assets/Prefabs/Player/PlayerProfile.prefab
#	Assets/Scenes/SpaceStation.unity
#	ProjectSettings/TagManager.asset

JPTFL committed:
merges puzzles_energycore in develop and integrates energy core task in space station scene

JPTFL committed:
Merge branch 'puzzles_storage_room' into develop_merge

# Conflicts:
#	Assets/Scenes/SpaceStation.unity
#	ProjectSettings/TagManager.asset

JPTFL committed:
merges puzzle_storage_room in develop_merge and integrates storage task in space station scene

JPTFL committed:
Merge branch 'puzzles_asteroidshooter' into develop_merge

# Conflicts:
#	Assets/Prefabs/Player/PlayerProfile.prefab
#	Assets/Scenes/SpaceStation.unity
#	Assets/Scripts/Game/Difficulty.cs
#	Assets/Scripts/PlayerController/PlayerProfileService.cs
#	ProjectSettings/TagManager.asset

JPTFL committed:
merges puzzle_asteroidshooter in develop_merge and integrates asteroid shooter task in space station scene
- refactors asteroid-shooter factory and game task to use new game structure

JPTFL committed:
adds missing tag manager tags for marble gravity task

JPTFL committed:
fixes wrong tags in unity editor for marble gravity labyrinth prefabs

JPTFL committed:
removes unused test scenes from merged task branches

JPTFL committed:
removes unused using-statements

JudiTeller committed:
remove HUD_Canvas from SpaceStation.unity

JudiTeller committed:
Merge branch 'develop_merge' into ui_progress

# Conflicts:
#	Assets/Prefabs/Player/PlayerProfile.prefab
#	Assets/Scripts/Game/Tasks/GameTaskFactory.cs

JPTFL committed:
Merge pull request #81 from JudiTeller/ui_progress

merges ui branch in manual develop_merge, to be integrated and merged in develop afterwards

Aremju reviewed (commented):
![image](https://github.com/JudiTeller/S5-VR-game/assets/98806292/651b51da-87ee-4d4f-a898-fb4a538abbf0)

JPTFL committed:
adds hud prefab instances to player prefab and integrates hud into game structure using GameHUDService

JPTFL committed:
reverts temporary timer value changes
- adds asteroid shooter task factory to the factory list again, after removing it for testing purpose

Aremju reviewed (approved):
Game is more and more looking like a game
